examples/embed/glfw/simple-window.py

```diff
@@ -0,0 +1,289 @@
+    global texture_data
+    global model_data
+    global program
+
+    global texture
+    global model
+    global color
+
+    data_path = data_path or _find_data_folder()
+
+    # Build file lists
+    files = os.listdir(data_path)
+    colors = list(itertools.product([0, 1], repeat=3))  # (0, 0, 0), (0, 0, 1), ... (1, 1, 1)
+    models = [Obj.open(os.path.join(data_path, filename)).pack() for filename in files if filename.endswith('.obj')]
+    textures = [Image.open(os.path.join(data_path, filename)) for filename in files if filename.endswith('.jpg')]
+    textures = [t for t in textures if t.mode == 'RGB']
+
+    # Setup global generators
+    color = _eternal_generator(colors)
+    texture = _eternal_generator(textures)
+    model = _eternal_generator(models)
+
+    # Setup keyboard control parameters
+    background_color = next(color)
+    foreground_color = next(color)
+    texture_image = next(texture)
+    model_data = next(model)
+
+    texture_data = ctx.texture(texture_image.size, 3, texture_image.tobytes())
+    texture_data.build_mipmaps()
+
+    # Setup files
+    vertex_path = vertex_path or f'{data_path}/shader.vert'
+    fragment_path = fragment_path or f'{data_path}/shader.frag'
+    object_path = object_path or f'{data_path}/mug.obj'
+
+    program = _load_shaders(ctx, vertex_path, fragment_path)
+
+
+def main_loop(data_path=None, texture_path=None, vertex_path=None, fragment_path=None, object_path=None):
+    global ctx
+
+    # Loop initialization
+    win = glfw_initialization()
+    ctx = ModernGL.create_context()
+    load_data(
+        ctx,
+        data_path=data_path,
+        vertex_path=vertex_path,
+        fragment_path=fragment_path,
+        texture_path=texture_path,
+        object_path=object_path,
+        )
+
+    # Display
+    while not glfw.window_should_close(win):
+        render_scene(win, ctx)
+        glfw.swap_buffers(win)
+        glfw.poll_events()
+
+    glfw.destroy_window(win)
+    glfw.terminate()
+
+
+def render_scene(win, ctx):
+    global fbo
+    global texture_data
+    global model_data
+    global program
+    global background_color
+    global foreground_color
+
+    # Matrices and Uniforms
+    perspective = pyrr.Matrix44.perspective_projection(45.0, 1.0, 0.1, 1000.0)
+    lookat = pyrr.Matrix44.look_at(
+        eye=(-85, -180, 140),
+        target=(0.0, 0.0, 65.0),
+        up=(0.0, 0.0, 1.0),
+        )
+    mvp = perspective * lookat
+
+    program.uniforms['Light'].value = (-140.0, -100.0, 350.0)
+    program.uniforms['Color'].value = tuple(list(foreground_color) + [0.25])
+    program.uniforms['Mvp'].write(mvp.astype('f4').tobytes())
+
+    # Vertex Buffer and Vertex Array
+    vbo = ctx.buffer(model_data)
+    vao = ctx.simple_vertex_array(program, vbo, ['in_vert', 'in_text', 'in_norm'])
+
+    ctx.disable(ModernGL.DEPTH_TEST)
+    ctx.clear(*background_color)
+
+    # check for window size change
+    fb_size = glfw.get_framebuffer_size(win)
+
+    # Framebuffers
```

Comment:
Creating a frambebuffer + renderbuffer + depth_renderbuffer every frame will eat up all your the memory.

Reply:
Cool.  I removed all of the fbo-specific code, including the screenshots.

I'm still noticing an uptick in both time and memory, though the slope of the increase is now far better; it looks like there may be a GLFW oddity related to buffer swaps.